hw/ip/hmac/data/hmac.hjson

```diff
@@ -169,7 +186,88 @@
                 ''',
           resval: "0",
         }
-      ]
+        { bits: "7:4",
+          name: "digest_size",
+          resval: "0x01",
+          desc: '''Digest size configuration.
+
+                This is a 4-bit one-hot field to select digest size for either HMAC or SHA-2.
+                Invalid values, i.e., values with multiple bits set and value 4'b0000 are
+                mapped in HMAC to SHA2_NONE (4'b0001).
```

Comment:
Nice, this seems familiar to me :-)
Here we should also specify what happens if that value is set once a decision has been taken.